### DISCUSSION TITLE
Use HxInputFile without using endpoint?

### DISCUSSION THREAD

**Discussion by resignationinfinite** [havit/Havit.Blazor#587] [Q&A] [answered]
I would like to use this component to upload a file using a service injected into the same page as the component rather than setting up a separate API endpoint on the server, is there a way to access the file stream itself in the HandleUploadClick method?

**Comment by rqx110** [comment]
see `OnChange`

**Comment by hakenr** [comment, marked as answer]
You can use the `HxInputFileCore` component which derives from `InputFile` and supports all the regular file-upload techniques described here: https://learn.microsoft.com/en-us/aspnet/core/blazor/file-uploads?view=aspnetcore-7.0

As mentioned by @rqx110, the `OnChange` callback is a good starting point. `OnChange` is also published by the `HxInputFile` which wraps the `HxInputFileCore` to a form-input with `Label` and such `HxInputXy` features.

**Reply by resignationinfinite** [reply]
Excellent, thank you so much, I have got it working now.
